### PR TITLE
updated filters, product page. new login-signup modal form

### PR DESCRIPTION
**Pull request by olka-velka**
I updated fragments for filters and product pages.
for login and sign up I tried modal form as a popup. it shows some conflict with scripts but I'll deal with that tomorrow or call the collective mind

### CODE REVIEW

File: src/main/java/com/southernartsoap/model/UserModel.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	@Column(name = "user_id")
+	private Long id;
+	
+	private String username;
+	private String firstName;
+	private String lastName;
+	private String email;
+	private String password;
+
```
Comment (jlangson) [inline src/main/java/com/southernartsoap/model/UserModel.java:33]: needs created_at timestamp

File: src/main/java/com/southernartsoap/model/UserModel.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package com.southernartsoap.model;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
```
Comment (jlangson) [inline src/main/java/com/southernartsoap/model/UserModel.java:8]: add imports for

import javax.validation.constraints.Email;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.Pattern;

File: src/main/java/com/southernartsoap/model/UserModel.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+public class UserModel {
+	
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	@Column(name = "user_id")
+	private Long id;
+	
+	private String username;
+	private String firstName;
+	private String lastName;
```
Comment (jlangson) [inline src/main/java/com/southernartsoap/model/UserModel.java:30]: add email input checks

@Length(min = 3, message = "Your username must have at least 3 characters")
    @Length(max = 15, message = "Your username cannot have more than 15 characters")
    @Pattern(regexp="[^\\s]+", message="Your username cannot contain spaces")

File: src/main/resources/templates/filters.html [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+							</div>
+						</a>
+					</div>
+				</div>
+			</div>
+
+
+
+			<div th:replace="fragments/footer"></div>
+			<script th:src="@{/webjars/jquery/jquery.min.js}"
```
Comment (jlangson) [inline src/main/resources/templates/filters.html:180]: remove scripts below fragment/footer

File: src/main/resources/templates/login-signup.html [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    </div>
+    <!--/.Content-->
+  </div>
+</div>
+	
+	
+	
+
+			<div th:replace="fragments/footer"></div>
+	<script th:src="@{/webjars/jquery/jquery.min.js}"
```
Comment (jlangson) [inline src/main/resources/templates/login-signup.html:149]: remove scripts below footer fragments

File: src/main/resources/templates/product.html [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+				</div>
+
+
+			</div>
+		</div>
+	</div>
+
+
+	<div th:replace="fragments/footer"></div>
+	<script th:src="@{/webjars/jquery/jquery.min.js}"
```
Comment (jlangson) [inline src/main/resources/templates/product.html:190]: remove scripts below footer fragment

**Review by jlangson — CHANGES_REQUESTED** [review]
See comments.

Remove unneeded scripts after footer fragment and add some tweaks to the UserModel.

**Review by julia-kim — APPROVED** [review]

**Pull request merged**